core/src/main/groovy/noe/rhel/server/httpd/HttpdBaseOS.groovy

```diff
@@ -0,0 +1,119 @@
+package noe.rhel.server.httpd
+
+import groovy.util.logging.Slf4j
+import noe.common.utils.Cmd
+import noe.common.utils.JBFile
+import noe.common.utils.Library
+import noe.common.utils.Version
+import noe.server.Httpd
+/**
+ * RHEL BaseOS Httpd server
+ * @author Jan Onderka <jonderka@redhat.com> *
+ */
+@Slf4j
+class HttpdBaseOS extends Httpd {
+  protected String serviceName
+
+  HttpdBaseOS(String basedir, version) {
+    super(basedir, version)
+    this.serviceName = "httpd"
+    this.basedir = "/etc/${serviceName}"
+
+    // please initialize in setDefault() method
+    setDefault()
+    logDirs.each { JBFile.makeAccessible(new File(getServerRoot() + it)) }
+  }
+
+  void setDefault() {
+    super.setDefault()
+    this.configDirs = ['/conf', '/conf.d']
+    if (version >= new Version("2.4")) {
+      this.configDirs.add('/conf.modules.d')
+    }
+    this.modulesDir = this.basedir + "/modules"
+    this.logDirs = ['/logs']
+    // what path will be taken as relative for star/stop cmd
+    this.binPath = '/'
+    if (platform.isRHEL6() || platform.isRHEL5() || platform.isRHEL4()) {
+      this.start = ['service', serviceName, 'start']  // starting as service
+      this.stop = "service ${serviceName} stop"  // stopping as service
+    } else {
+      this.start = ['systemctl', 'start', serviceName]
+      this.stop = ['systemctl', 'stop', serviceName]
+    }
+    this.apachectl = ['/usr/sbin/apachectl']
+    this.deploymentPath = "/var/www/html"
+    this.confDeploymentPath = this.basedir + '/conf.d'
+    this.cgiDeploymentPath = "/var/www/cgi-bin"
+    this.modClusterCacheDir = "/var/cache/mod_cluster"
+    this.opensslPath = 'openssl' // openssl is on $PATH on RHEL by OS installation
+    this.abPath = "/usr/bin/ab"
+    this.htdbmPath = "/usr/bin/htdbm"
+    this.htpasswdPath = "/usr/bin/htpasswd"
+    this.httxt2dbmPath = "/usr/bin/httxt2dbm"
+    this.logresolvePath = "/usr/bin/logresolve"
+
+    // ! please add new paths as absolute paths
+  }
+
+  File getPidFile() {
+    return new File("/var/run/${serviceName}/${serviceName}.pid")
+  }
+
+  Integer extractPid() {
+    def pidAsStr = new ByteArrayOutputStream()
+    if (Cmd.executeCommandRedirectIO("sudo cat ${getPidFile()}", null, null, pidAsStr, System.err) == 0) {
+      pidAsStr = pidAsStr.toString()
+      pidAsStr = pidAsStr.trim()
+      pidAsStr = pidAsStr.replaceAll('"', '')
```

Comment:
Can't we chain these?

`pidAsStr = pidAsStr.toString().trim().replaceAll('"','')`

Reply:
done